whois-query/src/main/java/net/ripe/db/whois/query/acl/HazelcastPersonalObjectAccounting.java

```diff
@@ -34,34 +35,43 @@ public HazelcastPersonalObjectAccounting(final HazelcastInstance hazelcastInstan
     public int getQueriedPersonalObjects(final InetAddress remoteAddress) {
         Integer count = null;
         try {
-            count = counterMap.get(remoteAddress);
+            count = remoteAddrCounterMap.get(remoteAddress);
         } catch (OperationTimeoutException | IllegalStateException e) {
             // no answer from hazelcast, expected, don't rethrow
             LOGGER.debug("{}: {}", e.getClass().getName(), e.getMessage());
         }
 
-        if (count == null) {
-            return 0;
+        return count == null ? 0 : count;
+    }
+
+    @Override
+    public int getQueriedPersonalObjects(final String ssoId) {
+        Integer count = null;
+        try {
+            count = ssoCounterMap.get(ssoId);
+        } catch (OperationTimeoutException | IllegalStateException e) {
+            // no answer from hazelcast, expected, don't rethrow
+            LOGGER.debug("{}: {}", e.getClass().getName(), e.getMessage());
         }
 
-        return count;
+        return count == null ? 0 : count;
```

Comment:
Instead of doing this, why not instanciate Integer count with a 0 instead a null?
Integer count = 0;
...
return count;

Reply:
Because by default Integer class has initilialization null and you cannot initialize 0 for all remoteAdress. so if entry is not available in map it will be returned null not zero